### ISSUE TITLE
Mapping over an `Arb` executes twice

### ISSUE THREAD

**Issue by Deeds67** [kotest/kotest#1389]
In kotest v4.0.3, the following code always prints twice, even though I'm only taking 1 value from the sequence:

```
val intArb = Arb.int(1, 10).map { i ->
                println("generated $i")
            }

intArb.values(RandomSource.Default).take(1).toList()
```

Output:
```
    generated 3
    generated 3
```

As well as using a custom arb:
```
val customArb = arb {
                generateSequence { 1 }
            }.map { i ->
                println("customArb $i")
            }

customArb.values(RandomSource.Default).take(1).toList()
```

Output:
```
    customArb 1
    customArb 1
```

**Comment by sksamuel** [comment]
The double usage is because it is mapping the initial value for the shrinker. I've changed this to be lazy so it will only call the map again if shrinking needs to take place, which for .take() it never will.

**Comment by LeoColman** [comment]
Closed via 5029d98300c9f733ba2d16120c81e5426468249c
